x/model/keeper/msg_server_model.go

```diff
@@ -64,6 +65,14 @@ func (k msgServer) CreateModel(goCtx context.Context, msg *types.MsgCreateModel)
 		model.CommissioningModeInitialStepsHint = 1
 	}
 
+	if model.EnhancedSetupFlowOptions == 0 {
+		model.EnhancedSetupFlowTCUrl = msg.EnhancedSetupFlowTCUrl
+		model.EnhancedSetupFlowTCRevision = msg.EnhancedSetupFlowTCRevision
+		model.EnhancedSetupFlowTCDigest = msg.EnhancedSetupFlowTCDigest
+		model.EnhancedSetupFlowTCFileSize = msg.EnhancedSetupFlowTCFileSize
```

Comment:
The `EnhancedSetupFlowTCFileSize` field can only be set if `EnhancedSetupFlowOptions` is present.